src/Schema/Exception/ForeignKeyAlreadyExists.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Schema\Exception;
+
+use Doctrine\DBAL\Schema\Name\OptionallyQualifiedName;
+use Doctrine\DBAL\Schema\SchemaException;
+use LogicException;
+
+use function sprintf;
+
+final class ForeignKeyAlreadyExists extends LogicException implements SchemaException
+{
+    public static function new(OptionallyQualifiedName $tableName, string $foreignKeyName): self
```

Comment:
`$foreignKeyName` could still be an `UnqualifiedName`. If it's auto-generated, you can convert it using `Unqualified::unquoted()`. In the future, auto-generated constraint names will be auto-generated by the underlying database, so this won't be necessary.